src/js/view/controls/components/menu/menu.js

```diff
@@ -294,51 +342,56 @@ export default class Menu extends Events {
         menu.destroy();
     }
     open(evt) {
-        if (this.visible && !this.openMenus) {
+        const mainMenuVisible = this.mainMenu.visible;
+        let focusEl;
+        if (!this.items.length || this.children) {
             return;
         }
-        backButtonTarget = null;
-        let focusEl;
         if (this.isSubmenu) {
+            const sourceEvent = evt.sourceEvent || event;
+            const firstItem = this.topbar ? this.topbar.firstChild : this.items[0].el;
+            const lastItem = this.items[this.items.length - 1].el;
+            const isKeydown = sourceEvent.type === 'keydown';
+            const key = isKeydown && sourceEvent.key.replace(/(Arrow|ape)/, '');
```

Comment:
this logic feels very similar to the key normalization done elsewhere. can we centralize somehow?

Reply:
I can make a util for normalizeKey that does the replace. Alternatively, can switch to keycodes (I usually prefer keyCodes but the old code uses this form of normalized key everywhere, and it does have the advantage of people not having to look them up so I opted to keep it).